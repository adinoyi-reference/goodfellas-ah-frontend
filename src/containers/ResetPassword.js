import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { resetPassword } from '../actions/authActions';
import AuthInput from '../components/shared/AuthInput';
import Button from '../components/shared/Button';
import validateAuth from '../lib/validation';
import Loading from '../components/shared/Loading';

const fieldNames = ['password', 'confirmPassword'];

const initialState = {
  password: '',
  confirmPassword: '',
  loading: false,
  touched: {
    password: false,
    confirmPassword: false,
  },
};

export class ResetPassword extends Component {
  state = initialState;

  handleResetPassword = async (e) => {
    e.preventDefault();
    const { resetPassword: newPassword, history } = this.props;
    const { password, confirmPassword } = this.state;
    const userData = {
      password,
      confirm_password: confirmPassword,
    };
    const validationError = validateAuth({ password, confirmPassword }, fieldNames);
    const changedTouchState = this.changeTouchState(fieldNames, true);
    this.setState({ touched: changedTouchState });

    if (!validationError.status) {
      this.setState({ loading: true });
      await newPassword(userData, history);
      this.setState(initialState);
    }
  };

  handleChange = (event) => {
    this.setState({ [event.target.id]: event.target.value });
  }

  handleBlur = (field) => () => {
    const { touched } = this.state;
    this.setState({
      touched: { ...touched, [field]: true },
    });
  }

  changeTouchState = (fields, state) => {
    const touchState = {};
    fields.forEach((field) => {
      touchState[field] = state;
    });
    return touchState;
  }

  render() {
    const { errorMessage } = this.props;
    const {
      password, confirmPassword, loading, touched,
    } = this.state;
    const validationError = validateAuth({ password, confirmPassword }, fieldNames);
    return (
      <form onSubmit={this.handleResetPassword} className="reset-form">
        <h3 className="reset-title text-center">Reset Password</h3>

        <AuthInput
          name="password"
          placeholder="New password"
          type="password"
          value={password}
          touched={touched}
          handleChange={this.handleChange}
          handleBlur={this.handleBlur('password')}
          error={validationError}
        />
        <AuthInput
          name="confirmPassword"
          placeholder="Confirm new password"
          type="password"
          value={confirmPassword}
          touched={touched}
          handleChange={this.handleChange}
          handleBlur={this.handleBlur('confirmPassword')}
          error={validationError}
        />
        <div className="error-field">{errorMessage}</div>
        {loading ? <button disabled="disabled" type="submit" className="auth-button loading"><Loading /></button> : <Button title="SUBMIT" className="auth-button" type="submit" />}

      </form>
    );
  }
}
const mapStateToProps = (state) => ({
  errorMessage: state.auth.errorMessage,
  successMessage: state.auth.successMessage,
});


export default connect(
  mapStateToProps,
  { resetPassword },
)(withRouter(ResetPassword));
