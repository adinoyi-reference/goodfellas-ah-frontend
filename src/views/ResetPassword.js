import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { resetPassword } from '../actions/authActions';

class ResetPassword extends Component {
  onSubmit = (e) => {
    e.preventDefault();
    const userData = {
      password: e.target.password.value,
      confirm_password: e.target.confirmPassword.value,
    };
    this.props.resetPassword(userData, this.props.history);
  };

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <input type="text" name="password" placeholder="input new password" />
        <input type="text" name="confirmPassword" placeholder="confirm new password" />
        <input type="submit" value="submit" />
      </form>
    );
  }
}
const mapStateToProps = (state) => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { resetPassword })(withRouter(ResetPassword));
