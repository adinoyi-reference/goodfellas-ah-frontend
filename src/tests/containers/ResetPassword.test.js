import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import ResetPassword from '../../containers/ResetPassword';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = shallow(

    <Root>
      <MemoryRouter>
        <ResetPassword />
      </MemoryRouter>
    </Root>,

  );
});
afterEach(() => wrapped.unmount());


describe('ResetPassword UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
