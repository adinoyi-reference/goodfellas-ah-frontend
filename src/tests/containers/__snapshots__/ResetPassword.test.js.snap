// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword UI render features container should render as expected 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          Object {
            "key": "testKey",
            "pathname": "/",
          },
        ]
      }
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "canGo": [Function],
            "createHref": [Function],
            "entries": Array [
              Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
            ],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "index": 0,
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <Connect(withRouter(ResetPassword))>
          <withRouter(ResetPassword)
            resetPassword={[Function]}
          >
            <Route>
              <ResetPassword
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/",
                        "search": "",
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/",
                    "search": "",
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                resetPassword={[Function]}
              >
                <form
                  onSubmit={[Function]}
                >
                  <AuthInput
                    error={
                      Object {
                        "password": "",
                      }
                    }
                    handleChange={[Function]}
                    name="password"
                    placeholder="New password"
                    type="password"
                  >
                    <div>
                      <label
                        className="sr-only"
                        htmlFor="password"
                      >
                        password
                      </label>
                      <input
                        className="auth-field"
                        id="password"
                        onChange={[Function]}
                        placeholder="New password"
                        type="password"
                      />
                      <div
                        className="error-field"
                      />
                    </div>
                  </AuthInput>
                  <AuthInput
                    error={
                      Object {
                        "confirmPassword": undefined,
                      }
                    }
                    handleChange={[Function]}
                    name="confirmPassword"
                    placeholder="Confirm new password"
                    type="password"
                  >
                    <div>
                      <label
                        className="sr-only"
                        htmlFor="confirmPassword"
                      >
                        confirmPassword
                      </label>
                      <input
                        className="auth-field"
                        id="confirmPassword"
                        onChange={[Function]}
                        placeholder="Confirm new password"
                        type="password"
                      />
                      <div
                        className="error-field"
                      />
                    </div>
                  </AuthInput>
                  <AuthButton
                    name="Submit"
                  >
                    <div>
                      <button
                        className="auth-button"
                        type="submit"
                      >
                        Submit
                      </button>
                    </div>
                  </AuthButton>
                </form>
              </ResetPassword>
            </Route>
          </withRouter(ResetPassword)>
        </Connect(withRouter(ResetPassword))>
      </Router>
    </MemoryRouter>
  </Provider>
</_default>
`;
